plugins/modules/identity/keycloak/keycloak_group.py

```diff
@@ -217,30 +228,41 @@ def main():
         realm=dict(default='master'),
         id=dict(type='str'),
         name=dict(type='str'),
-        attributes=dict(type='dict')
+        attributes=dict(type='dict'),
+        token=dict(type='str'),
     )
 
     argument_spec.update(meta_args)
 
     module = AnsibleModule(argument_spec=argument_spec,
                            supports_check_mode=True,
-                           required_one_of=([['id', 'name']]))
+                           required_one_of=([['id', 'name'],
+                                             ['token', 'auth_realm', 'auth_username', 'auth_password']]),
+                           required_together=([['auth_realm', 'auth_username', 'auth_password']]))
 
     result = dict(changed=False, msg='', diff={}, group='')
 
     # Obtain access token, initialize API
-    try:
-        connection_header = get_token(
-            base_url=module.params.get('auth_keycloak_url'),
-            validate_certs=module.params.get('validate_certs'),
-            auth_realm=module.params.get('auth_realm'),
-            client_id=module.params.get('auth_client_id'),
-            auth_username=module.params.get('auth_username'),
-            auth_password=module.params.get('auth_password'),
-            client_secret=module.params.get('auth_client_secret'),
-        )
-    except KeycloakError as e:
-        module.fail_json(msg=str(e))
+    token = module.params.get('token')
+    if(token is not None):
+        connection_header = {
+            'Authorization': token,
+            'Content-Type': 'application/json'
+        }
+    else:
+        try:
+            connection_header = get_token(
+                base_url=module.params.get('auth_keycloak_url'),
+                validate_certs=module.params.get('validate_certs'),
+                auth_realm=module.params.get('auth_realm'),
+                client_id=module.params.get('auth_client_id'),
+                auth_username=module.params.get('auth_username'),
+                auth_password=module.params.get('auth_password'),
+                client_secret=module.params.get('auth_client_secret'),
+            )
```

Comment:
It might be better to let `get_token` also handle the token parameter. Then you have to modify less if you want to add token support to all modules, since you just have to pass on `module.params['token']` as well (the `get()` is not necessary for all these values).

It might also be a good idea to refactor `get_token` so you can just pass it `module` or `module.params`, and it will extract the parameters itself. That will safe a lot of copy'n'paste between keycloak modules...

Reply:
Good idea, I will implement it as soon as possible